### PR TITLE
Adding test for target_teams_distribute_parallel_for_if with parallel modifier

### PR DESCRIPTION
**Pull request by tmh97**
Passes with GCC 11

Addresses issue #81

### CODE REVIEW

File: tests/4.5/target_teams_distribute_parallel_for/test_target_teams_distribute_parallel_for_if_parallel_modifier.F90 [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+   OMPVV_REPORT_AND_RETURN()  
+
+CONTAINS 
+   LOGICAL FUNCTION checkPreconditions()
+      LOGICAL :: isOffloading
+      INTEGER :: i, raiseWarningDevice, raiseWarningHost
+      INTEGER, DIMENSION(N) :: init_num_threads_dev, init_num_threads_host
+
+      OMPVV_TEST_AND_SET_OFFLOADING(isOffloading)
+      OMPVV_WARNING_IF(isOffloading .eqv. .FALSE. , "With offloading off, it is not possible to test if code executes on parallel or target")
```
Comment (spophale) [inline tests/4.5/target_teams_distribute_parallel_for/test_target_teams_distribute_parallel_for_if_parallel_modifier.F90:45]: on parallel ?

**Comment by seyonglee** [comment]
The behavior of the Fortran version differs from that of the C version; in C version, the compute region is offloaded (NUM_ATTEMPTS - ATTEMPT_THRESHOLD) times, while the Fortran version offloads the compute region (NUM_ATTEMPTS - ATTEMPT_THRESHOLD + 1) times. If we want to the current behavior of the Fortran version, we will have to change line 130 from "(NUM_ATTEMPTS - ATTEMPT_THRESHOLD)" to "(NUM_ATTEMPTS - ATTEMPT_THRESHOLD+1)".

**Comment by tmh97** [comment]
> The behavior of the Fortran version differs from that of the C version; in C version, the compute region is offloaded (NUM_ATTEMPTS - ATTEMPT_THRESHOLD) times, while the Fortran version offloads the compute region (NUM_ATTEMPTS - ATTEMPT_THRESHOLD + 1) times. If we want to keep the current behavior of the Fortran version, we will have to change line 130 from "(NUM_ATTEMPTS - ATTEMPT_THRESHOLD)" to "(NUM_ATTEMPTS - ATTEMPT_THRESHOLD+1)". If we want the Fortran version has the same behavior as the C version, we will have to do the following changes: in line 101: change "attempt .ge. ATTEMPT_THRESHOLD" to "attempt .gt. ATTEMPT_THRESHOLD" in line 108: change "attempt .ge. ATTEMPT_THRESHOLD" to "attempt .gt. ATTEMPT_THRESHOLD" in line 124: change "a(i) .ne. ATTEMPT_THRESHOLD - 1" to "a(i) .ne. ATTEMPT_THRESHOLD"

@seyonglee thanks again for your help! My most recent commit addressed this issue

File: tests/4.5/target_teams_distribute_parallel_for/test_target_teams_distribute_parallel_for_if_parallel_modifier.F90 (hunk 1 above)
Comment (tmh97) [reply tests/4.5/target_teams_distribute_parallel_for/test_target_teams_distribute_parallel_for_if_parallel_modifier.F90]: removed print statement altogether

**Review by seyonglee — APPROVED** [review]

**Review by nolanbaker31 — APPROVED** [review]

**Pull request merged**